src/box/vy_log.c

```diff
@@ -1026,6 +1026,51 @@ vy_log_bootstrap(void)
 	return 0;
 }
 
+/**
+ * Return true if the last vylog is new and contains no user data
+ * (i.e. last entry is VY_LOG_SNAPSHOT).
+ * In case of any errors log them and return false.
+ */
+static bool
+vy_log_last_entry_is_snapshot(void)
+{
+	const char *path =
+		vy_log_filename(vclock_sum(&vy_log.last_checkpoint));
+	if (access(path, F_OK) < 0) {
+		say_error("Failed to access last vylog");
+		return false;
+	}
+	struct xlog_cursor cursor;
+	if (xdir_open_cursor(&vy_log.dir,
+			     vclock_sum(&vy_log.last_checkpoint),
+			     &cursor) < 0) {
+		diag_log();
+		diag_clear(diag_get());
+		return false;
+	}
+	int rc;
+	struct xrow_header row;
+	while ((rc = xlog_cursor_next(&cursor, &row, false)) == 0) {
+		struct vy_log_record record;
+		rc = vy_log_record_decode(&record, &row);
+		if (rc < 0)
+			break;
+		if (record.type == VY_LOG_SNAPSHOT) {
```

Comment:
Can it contain more than one `VY_LOG_SNAPSHOT`?

Reply:
No, it can't: the only place where VY_LOG_SNAPSHOT is added - during vylog rotation (after all squashed records are written to the new file).